python_agent/test_custom_env.py

```
@@ -0,0 +1,154 @@
+
+        self.mock_zmq_client.execute_action.reset_mock() # Reset call count
+        _, reward, _, _, _ = self.env.step(5) # Action 5 = pickup_bone
+        
+        self.mock_zmq_client.execute_action.assert_not_called()
+        self.assertEqual(reward, -1.0) # Penalty for trying to pick up non-visible bones
+
+    def test_step_pickup_bone_bones_visible(self):
+        # Ensure last_observation is set and has bones
+        obs_with_bones = self._get_default_raw_obs()
+        obs_with_bones["nearby_ground_items"] = [
+            {"id": BONE_ITEM_ID, "name": "Bones", "quantity": 1, "location": {"x": 3202, "y": 3202, "plane": 0}}
```

Ruff (0.11.9)

Trailing comma missing

Add trailing comma

(COM812)